### PR TITLE
Add Hedera RPC reference page

### PR DESCRIPTION
**Pull request by huck-hgraph**
### Summary
Adds RPC Reference page for Hedera with documentation of the following JSON RPC methods:
- `hedera_signAndExecuteTransaction`
- `hedera_signAndReturnTransaction`
- `hedera_signMessage`

### CODE REVIEW

File: docs/blockchain-rpc/hedera-rpc.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
+---
+description: Hedera JSON-RPC Methods
+---
+
+# Hedera
+
```
Comment (huck-hgraph) [inline docs/blockchain-rpc/hedera-rpc.md:6]: @HGraphPunks @hgraphql LMK what you think of this intro paragraph. I feel like it could be better, but just wanted to get something up to start the discussion.

Comment (hgraphql) [reply docs/blockchain-rpc/hedera-rpc.md:6]: This is good. I think maybe reword in the affirmative, something like:

"The following JSON-RPC methods are for native integration into Hedera." "Hedera also has an open-source project implementing the [Ethereum JSON-RPC standard](https://docs.hedera.com/hedera/core-concepts/smart-contracts/json-rpc-relay) which is not covered in this documentation"

I think there's opportunity in the future to test and integrate this relay into WalletConnect, to provide access to a wider audience of dApp providers that are familiar with the Ethereum JSON-RPC API

Comment (huck-hgraph) [reply docs/blockchain-rpc/hedera-rpc.md:6]: Ooh nice wording. I'll update

Comment (huck-hgraph) [reply docs/blockchain-rpc/hedera-rpc.md:6]: Updated:
Screenshot 2023-08-04 at 2 28 38 PM

File: docs/blockchain-rpc/hedera-rpc.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+The `hedera_signAndExecuteTransaction` method is a generic method for sending a transaction that has been converted to bytes to the wallet for signing and execution. The dApp can therefore construct any type of Hedera transaction with the [`hedera-sdk-js` library](https://github.com/hashgraph/hedera-sdk-js)
+
+### Parameters
+
+    1. `Object` - signAndExecuteTransaction parameters
+      1.1 `transaction` : `Object` - transaction parameters
+        1.1.1 `type` : `String` - The Hedera function name (*1)
+        1.1.2 `bytes` : `String` - Transaction that has been converted to bytes array (*2) and encoded as a base64 string.
+
+\*1. Full list of functions can be found [here](https://hashgraph.github.io/hedera-protobufs/#proto.HederaFunctionality), and are available via the [`RequestType` class](https://github.com/hashgraph/hedera-sdk-js/blob/50d7f76ffd1fd797a029192fab08b035720998b9/src/RequestType.js) from the Hedera SDK.
```
Comment (hgraphql) [inline docs/blockchain-rpc/hedera-rpc.md]: thoughts on referencing the most upstream branch? (develop in this case)

`https://github.com/hashgraph/hedera-sdk-js/blob/develop/src/RequestType.js`

My thoughts are if it changes or moves, this link would be stale anyways. Though maybe that's the time to update

Comment (hgraphql) [inline docs/blockchain-rpc/hedera-rpc.md]: `A full list` or `The full list`?

File: docs/blockchain-rpc/hedera-rpc.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+transaction to the Hedera network, the wallet just signs the transaction and returns the signed transaction to the dApp for further handling.
+
+### Parameters
+
+    1. `Object` - signAndExecuteTransaction parameters
+      1.1 `transaction` : `Object` - transaction parameters
+        1.1.1 `type` : `String` - The Hedera function name (*1)
+        1.1.2 `bytes` : `String` - Transaction that has been converted to bytes array (*2) and encoded as a base64 string
+
+\*1. Full list of functions can be found [here](https://hashgraph.github.io/hedera-protobufs/#proto.HederaFunctionality), and are available via the [`RequestType` class](https://github.com/hashgraph/hedera-sdk-js/blob/50d7f76ffd1fd797a029192fab08b035720998b9/src/RequestType.js) from the Hedera SDK.
```
Comment (hgraphql) [inline docs/blockchain-rpc/hedera-rpc.md]: `The` or `a` here too?

File: docs/blockchain-rpc/hedera-rpc.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      "type": "CryptoTransfer",
+      "bytes": "VGhpcyBpcyBqdXN0IHNvbWUgc3R1..."
+    }
+  }
+}
+```
+
+## hedera_signMessage
+
+The `hedera_signAndReturnTransaction` method is used to pass arbitrary data to the wallet to be signed with the user's private key. The resulting signed message is then passed back to the dApp.
```
Comment (hgraphql) [inline docs/blockchain-rpc/hedera-rpc.md]:
```suggestion
The `hedera_signMessage` method is used to pass arbitrary data to the wallet to be signed with the user's private key. The resulting signed message is then passed back to the dApp.
```

**Review by hgraphql — APPROVED** [review]
lgtm with a few comments

File: docs/blockchain-rpc/hedera-rpc.md (hunk 2 above)
Comment (huck-hgraph) [reply docs/blockchain-rpc/hedera-rpc.md]: Good catch. Didn't realize it wasn't already doing this.

**Comment by huck-hgraph** [comment]
@hgraphql do you think we should merge to our forked main and then open PR to walletconnect/main, or just change the target branch of this PR? Might be nice to squash and merge here first.

**Comment by hgraphql** [comment]
> @hgraphql do you think we should merge to our forked main and then open PR to walletconnect/main, or just change the target branch of this PR? Might be nice to squash and merge here first.

squash and then PR works for me
